src/ActivitiesSDKAdapter.js

```diff
@@ -191,22 +210,50 @@ export default class ActivitiesSDKAdapter extends ActivitiesAdapter {
    */
   postActivity(activity) {
     logger.debug('ACTIVITY', undefined, 'postActivity()', ['called with', {activity}]);
-    const hasCards = this.hasAdaptiveCards(activity);
-
-    const object = !hasCards ? activity.text : {
-      cards: activity.cards.map((card) => JSON.stringify(card)),
-      displayName: activity.text,
-    };
-
-    const {id, cluster = 'us'} = deconstructHydraId(activity.roomID);
-
-    return from(this.datasource.internal.conversation.post(
-      {
-        id,
-        cluster,
-      },
-      object,
-    )).pipe(
+
+    const mutableActivity = activity;
+
+    const activity$ = this.hasAdaptiveCards(activity)
+      ? from(this.fetchConversation(activity.roomID)).pipe(
```

Comment:
let's move this into a dedicated `encryptCards()` function and refactor the structure of `postActivity()` for simplicity:

```js
const doPost = async ()  => {
  const {id, cluster = 'us'} = deconstructHydraId(activityToCreate.roomID);
  const hasCards = this.hasAdaptiveCards(activity);
  const object = hasCards ?
      ({cards: await encryptCards(activity.cards), activityText: activity.text}) :
      activity.Text;

  return this.datasource.internal.conversation.post({id, cluster}, object);
}

return defer(doPost()).pipe(
  catchError(...could not post activity... throw err);
);